internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -250,6 +248,25 @@ func resourceKustoCluster() *pluginsdk.Resource {
 
 			"tags": commonschema.Tags(),
 		},
+
+		CustomizeDiff: pluginsdk.CustomDiffWithAll(
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				oldSubnetId, newSubnetId := d.GetChange("virtual_network_configuration.0.subnet_id")
+				if oldSubnetId != "" && newSubnetId != "" && oldSubnetId != newSubnetId {
```

Comment:
```suggestion
				if oldSubnetId != newSubnetId {
```
I believe this condition can be simplified